include/re_http.h

```diff
@@ -114,16 +115,20 @@ int  http_msg_print(struct re_printf *pf, const struct http_msg *msg);
 struct http_cli;
 struct http_req;
 struct dnsc;
+struct tcp_conn;
+struct tls_conn;
 
 typedef void (http_resp_h)(int err, const struct http_msg *msg, void *arg);
-typedef void (http_data_h)(struct mbuf *mb, void *arg);
+typedef int  (http_data_h)(const uint8_t *buf, size_t size,
+			   const struct http_msg *msg, void *arg);
+typedef void (http_conn_h)(struct tcp_conn *tc, struct tls_conn *sc,
+			   void *arg);
```

Comment:
perhaps add a doxygen comment to the prototypes, documenting the semantics of the
callback handler and the arguments.

Reply:
That make sense, but as they were not commented initially, let's do that in a separate PR.

Reply:
The connection handler was added to provide better support for applications that needs
to continue communication on the existing TCP connection after the HTTP response is received. Examples include web-socket clients and streaming clients. If this handler is set, the HTTP client will not reuse this connection in new HTTP requests.

Reply:
[HTTP/2](https://tools.ietf.org/html/rfc7540#section-3.2) is another example where this is useful.